### PR TITLE
Pass message metadata to interpreter when parsing

### PR DESCRIPTION
**Pull request by pheel**
**Proposed changes**:
- Currently a NaturalLanguageInterpreter's parse method receives message text, id and the tracker. This adds an optional argument for the message's metadata as well, and thus allows metadata to affect interpretation.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @tttthomasssss will take a look at it as soon as possible ✨

**Comment by pheel** [comment]
@federicotdn Would you mind having a look at this one? 🙏

**Comment by federicotdn** [comment]
Hey @pheel, could you share a bit more context on why you need the metadata for `NaturalLanguageInterpreter.parse`? Are you trying to do this on a custom subclass?

**Comment by pheel** [comment]
@federicotdn Exactly right! this is for a custom class, which makes use of contextual metadata sent over a socket to guide interpretation.

Right now an incoming message `m`'s metadata is written into the tracker _after_ interpretation, at the same time as the parse data that results from interpretation. Since `NaturalLanguageInterpreter` is given access to the pre-interpretation tracker state, containing all information about messages that came before `m`, I think it should have access to at least as much information about `m` itself.

File: changelog/6275.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add optional metadata argument to NaturalLanguageInterpreter's parse method.
```
Comment (federicotdn) [inline changelog/6275.feature.rst:1]:
```suggestion
Add optional metadata argument to ``NaturalLanguageInterpreter``'s parse method.
```

**Review by federicotdn — APPROVED** [review]
Ok! Looks good. You you mind merging master into it? I think there might be some merge conflicts.

**Review by federicotdn — APPROVED** [review]
Thanks!

**Pull request merged**